ts/src/test/scala/ch/epfl/ts/test/data/StrategyParameters.scala

```diff
@@ -18,41 +26,89 @@ class StrategyParametersTests extends FunSuite {
    */
   val legalCurrencies1 = (Currency.EUR, Currency.CHF)
   val legalCurrencies2 = (Currency.GBP, Currency.USD)
-  val illlegalCurrencies = (Currency.GBP, Currency.GBP)
+  val illegalCurrencies = (Currency.GBP, Currency.GBP)
   val legalCoefficient = 0.2342341341
   val illegalCoefficient = -0.2342341341
-  
+
   private type CurrencyPair = (Currency.Currency, Currency.Currency)
-  
+
   /** Generic function to test basic functionality expected from Parameter subclasses */
-  def testConcreteParameter[T](parameterTrait: ParameterTrait[T], validValues: Iterable[T], invalidValues: Iterable[T]) = {
+  def testConcreteParameter[V](parameterTrait: ParameterTrait{ type T = V}, validValues: Iterable[V], invalidValues: Iterable[V]) = {
+
     test(parameterTrait + " should give back the value it was instantiated with") {
       for {
         myValue <- validValues
-        parameter = parameterTrait.getInstance(myValue)        
+        parameter = parameterTrait.getInstance(myValue)
       } yield assert(parameter.value() == myValue, "Should give back value " + myValue)
     }
-    
+
     test(parameterTrait + " should reject invalid values") {
       for {
         myValue <- invalidValues
         attempt = Try(parameterTrait.getInstance(myValue))
       } yield assert(attempt.isFailure, "Should fail with illegal value " + myValue)
     }
-  
-    test(parameterTrait + " should accept all values within the acceptable range") {
+
+    test(parameterTrait + " should accept the 100 first values it declares as valid values") {
       for {
-        value <- parameterTrait.validValues
+        value <- (parameterTrait.validValues take 100).toList
         attempt = Try(parameterTrait.getInstance(value))
       } yield assert(attempt.isSuccess, "Should accept " + value)
     }
-    
+
     test(parameterTrait + " should have a default value which is valid") {
     	val attempt = Try(parameterTrait.getInstance(parameterTrait.defaultValue))
     	assert(attempt.isSuccess, "Should accept " + parameterTrait.defaultValue)
     }
   }
-  
+
+
+  /**
+   * Generic function to test concrete trading strategy implementation's correct
+   * behavior when instantiated with correct & incorrect parameters
+   */
+  def testConcreteStrategy(strategyCompanion: TraderCompanion) = {
+    implicit val builder = new ComponentBuilder("ConcreteStrategyTest")
+    
+    val traderId = 42L
+    def make(p: StrategyParameters) = strategyCompanion.getInstance(traderId, p, "TraderBeingTested")
```

Comment:
If you're going to pass the builder implicitly, please make sure that if you don't provide it you'll get some reasonable error.